pkg/operator/ceph/controller/predicate.go

```diff
@@ -83,6 +83,9 @@ func WatchControllerPredicate() predicate.Funcs {
 				} else if objectToBeDeleted(objOld, objNew) {
 					logger.Debugf("CR %q is going be deleted", objNew.Name)
 					return true
+				} else if FinalizerAdded(objOld, objNew) {
+					logger.Infof("finalizer added to object store %q, proceeding with reconcile", objNew.Name)
+					return true
 				} else if objOld.GetGeneration() != objNew.GetGeneration() {
 					logger.Debugf("skipping resource %q update with unchanged spec", objNew.Name)
 				}
```

Comment:
Is the `FinalizerAdded()` check necessary? You said that applying a finalizer changes the generation, which will mean it is reconciled with the condition below, right?

Reply:
Oh good point, I'll test without that `FinalizerAdded()` check and confirm that the generation change already covers it.